tests/utils/spline_builder_deriv_field_2d_test.cpp

```diff
@@ -0,0 +1,434 @@
+        DerivField<double, IdxRange<DerivX, GridXH, DerivY, GridYH>, MemorySpace>
+                function_and_derivs,
+        DField<IdxRange<DerivX, GridYH>> derivs_xmin,
+        DField<IdxRange<DerivX, GridYH>> derivs_xmax,
+        DField<IdxRange<GridXH, DerivY>> derivs_ymin,
+        DField<IdxRange<GridXH, DerivY>> derivs_ymax,
+        DField<IdxRange<DerivX, DerivY>> derivs_xy_min_min,
+        DField<IdxRange<DerivX, DerivY>> derivs_xy_max_min,
+        DField<IdxRange<DerivX, DerivY>> derivs_xy_min_max,
+        DField<IdxRange<DerivX, DerivY>> derivs_xy_max_max)
+{
+    IdxRangeSlice<GridXH> idx_range_slice_dx
+            = function_and_derivs.template idx_range_for_deriv<GridXH>();
+    IdxRangeSlice<GridYH> idx_range_slice_dy
+            = function_and_derivs.template idx_range_for_deriv<GridYH>();
+
+    IdxRange<GridXH> idx_range_x(get_idx_range(derivs_ymin));
+    IdxRange<GridYH> idx_range_y(get_idx_range(derivs_xmin));
+
+    Coord<X> x_min(ddc::coordinate(idx_range_x.front()));
+    Coord<X> x_max(ddc::coordinate(idx_range_x.back()));
+    Coord<Y> y_min(ddc::coordinate(idx_range_y.front()));
+    Coord<Y> y_max(ddc::coordinate(idx_range_y.back()));
+
+    Idx<DerivX> first_dx(1);
+    Idx<DerivY> first_dy(1);
+
+    Idx<GridXH> idx_slice_xmin(idx_range_slice_dx.front());
+    Idx<GridXH> idx_slice_xmax(idx_range_slice_dx.back());
+    Idx<GridYH> idx_slice_ymin(idx_range_slice_dy.front());
+    Idx<GridYH> idx_slice_ymax(idx_range_slice_dy.back());
+
+    Idx<DerivX, GridXH> idx_deriv_xmin(first_dx, idx_slice_xmin);
+    Idx<DerivX, GridXH> idx_deriv_xmax(first_dx, idx_slice_xmax);
+    Idx<DerivY, GridYH> idx_deriv_ymin(first_dy, idx_slice_ymin);
+    Idx<DerivY, GridYH> idx_deriv_ymax(first_dy, idx_slice_ymax);
+
+    Idx<DerivX, GridXH, DerivY, GridYH> idx_cross_deriv_min_min(idx_deriv_xmin, idx_deriv_ymin);
+    Idx<DerivX, GridXH, DerivY, GridYH> idx_cross_deriv_max_min(idx_deriv_xmax, idx_deriv_ymin);
+    Idx<DerivX, GridXH, DerivY, GridYH> idx_cross_deriv_min_max(idx_deriv_xmin, idx_deriv_ymax);
+    Idx<DerivX, GridXH, DerivY, GridYH> idx_cross_deriv_max_max(idx_deriv_xmax, idx_deriv_ymax);
+
+    ddc::parallel_for_each(
+            ExecSpace(),
+            idx_range_y,
+            KOKKOS_LAMBDA(Idx<GridYH> const idx_y) {
+                double const y = ddc::coordinate(idx_y);
+                derivs_xmin(first_dx, idx_y) = -2. / 3 * M_PI
+                                               * Kokkos::sin(2. / 3 * M_PI * double(x_min) + 0.25)
+                                               * Kokkos::sin(y);
+                derivs_xmax(first_dx, idx_y) = -2. / 3 * M_PI
+                                               * Kokkos::sin(2. / 3 * M_PI * double(x_max) + 0.25)
+                                               * Kokkos::sin(y);
+                function_and_derivs[idx_deriv_xmin](idx_y) = derivs_xmin(first_dx, idx_y);
+                function_and_derivs[idx_deriv_xmax](idx_y) = derivs_xmax(first_dx, idx_y);
```

Comment:
Why do you slice before indexing? Can't you just do:
```suggestion
                function_and_derivs(idx_deriv_xmin, idx_y) = derivs_xmin(first_dx, idx_y);
                function_and_derivs(idx_deriv_xmax, idx_y) = derivs_xmax(first_dx, idx_y);
```
?

Reply:
```
/src/data_types/derivative_field.hpp: In instantiation of ‘constexpr DerivField<ElementType, ddc::DiscreteDomain<ODDims ...>, MemorySpace, LayoutStridedPolicy>::reference DerivField<ElementType, ddc::DiscreteDomain<ODDims ...>, MemorySpace, LayoutStridedPolicy>::operator()(DElem ...) const [with DElem = {ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH>, ddc::DiscreteElement<{anonymous}::GridYH>}; ElementType = double; DDims = {ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, ddc::Deriv<{anonymous}::Y>, {anonymous}::GridYH}; MemorySpace = Kokkos::HostSpace; LayoutStridedPolicy = Kokkos::layout_right; reference = double&]’:
/tests/utils/spline_builder_deriv_field_2d_test.cpp:134:36:   required from here
/src/data_types/derivative_field.hpp:374:25: error: no matching function for call to ‘detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >::Combine(ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH>&, ddc::DiscreteElement<{anonymous}::GridYH>&)’
  374 |         full_index_type elem(elems...);
      |                         ^~~~
In file included from /home/pavi/CppProject/copy_voicexx/gyselalibxx_github_2/src/data_types/derivative_field_common.hpp:10,
                 from /home/pavi/CppProject/copy_voicexx/gyselalibxx_github_2/src/data_types/derivative_field.hpp:7:
/src/data_types/deriv_details.hpp:197:8: note: candidate: ‘constexpr detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >::Combine()’
  197 | struct Combine<Container>
      |        ^~~~~~~~~~~~~~~~~~
/src/data_types/deriv_details.hpp:197:8: note:   candidate expects 0 arguments, 2 provided
/src/data_types/deriv_details.hpp:197:8: note: candidate: ‘constexpr detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >::Combine(const detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >&)’
/src/data_types/deriv_details.hpp:197:8: note:   candidate expects 1 argument, 2 provided
/src/data_types/deriv_details.hpp:197:8: note: candidate: ‘constexpr detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >::Combine(detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >&&)’
/src/data_types/deriv_details.hpp:197:8: note:   candidate expects 1 argument, 2 provided
/src/data_types/derivative_field.hpp:375:45: error: no matching function for call to ‘DerivField<double, ddc::DiscreteDomain<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, ddc::Deriv<{anonymous}::Y>, {anonymous}::GridYH>, Kokkos::HostSpace>::get_internal_field(full_index_type&) const’
  375 |         return base_type::get_internal_field(elem)();
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/src/data_types/derivative_field_common.hpp:325:26: note: candidate: ‘template<class ... ODims> auto DerivFieldCommon<FieldType, ddc::DiscreteDomain<ODDims ...> >::get_internal_field(ddc::DiscreteDomain<ODims ...>) const [with ODims = {ODims ...}; FieldType = ddc::ChunkSpan<double, ddc::DiscreteDomain<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, ddc::Deriv<{anonymous}::Y>, {anonymous}::GridYH>, Kokkos::layout_right, Kokkos::HostSpace>; DDims = {ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, ddc::Deriv<{anonymous}::Y>, {anonymous}::GridYH}]’
  325 |     KOKKOS_FUNCTION auto get_internal_field(IdxRange<ODims...> idx_range) const
      |                          ^~~~~~~~~~~~~~~~~~
/src/data_types/derivative_field_common.hpp:325:26: note:   template argument deduction/substitution failed:
/src/data_types/derivative_field.hpp:375:45: note:   ‘detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >’ is not derived from ‘ddc::DiscreteDomain<Tags ...>’
  375 |         return base_type::get_internal_field(elem)();
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/src/data_types/derivative_field_common.hpp:383:26: note: candidate: ‘template<class ... ODims> auto DerivFieldCommon<FieldType, ddc::DiscreteDomain<ODDims ...> >::get_internal_field(ddc::DiscreteElement<QueryDDims ...>) const [with ODims = {ODims ...}; FieldType = ddc::ChunkSpan<double, ddc::DiscreteDomain<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, ddc::Deriv<{anonymous}::Y>, {anonymous}::GridYH>, Kokkos::layout_right, Kokkos::HostSpace>; DDims = {ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, ddc::Deriv<{anonymous}::Y>, {anonymous}::GridYH}]’
  383 |     KOKKOS_FUNCTION auto get_internal_field(Idx<ODims...> elem) const
      |                          ^~~~~~~~~~~~~~~~~~
/src/data_types/derivative_field_common.hpp:383:26: note:   template argument deduction/substitution failed:
/src/data_types/derivative_field.hpp:375:45: note:   ‘detail::Combine<ddc::DiscreteElement<ddc::Deriv<{anonymous}::X>, {anonymous}::GridXH, {anonymous}::GridYH> >’ is not derived from ‘ddc::DiscreteElement<Tags ...>’
  375 |         return base_type::get_internal_field(elem)();
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
```

Reply:
Ah no, it is just due to a missing `typename`. Fixed : 77d27150b